### ISSUE TITLE
Delete use case for Dependency editor's insights

### ISSUE THREAD

**Issue by arunkumars08** [openshiftio/openshift.io#3874]
Currently as shown in the below image, dependency editor's insights section handles removal of dependencies only from UI end. (If the same is refreshed, the deleted ones are sure to show up back)

This task can be used for the discussion around the implementation of it in the backend and the feasibilities.

What I can think for now,

1. Needs user mapping (as if one dependency when deleted by a user shouldn't be deleted across)
2. Needs stack mapping (the same user might find a dependency 'A' to be useful in stack 'S1' while not in 'S2'

Please feel free to add more thoughts on this.

<img width="836" alt="screen shot 2018-06-28 at 2 27 24 pm" src="https://user-images.githubusercontent.com/7880465/42024374-aab1b284-7adf-11e8-9098-a3fa6568c4cc.png">

**Comment by arunkumars08** [comment]
The easier solution is to not show the 'Remove' icon.

**Comment by slemeur** [comment]
Not show the "remove" icon to avoid the proposal to be removed?

**Comment by arunkumars08** [comment]
@slemeur Remove is a very nice feature to have and I am always on the same side. Currently, given the huge architectural changes that might be required to address this feature in the backend, it would be wise to not show it until we are ready.

Your thoughts are welcome.

**Comment by slemeur** [comment]
Maybe we can simplify.

We want the user to be able to hide certain recommendations will he is doing the setup of a project. The user should be able to tell to the system "I'm not interested by this library, don't show it anymore while I continue to setup this project". 
While this is true for the session, for the unique project the user is setting-up, it might not be true for the next session (when he'll setup another project). If the user is setting-up another project, the recommendation could be interesting for that new project. 

Does it help?

**Comment by arunkumars08** [comment]
Yes. if I understand your point correctly, the suggestion is to use that as a "hide for now" feature rather than an 'entirely remove' feature.

Which means, if a user clicks on the hide, it gets hidden in the current view but when he/she refreshes it is ok if it shows up.

**Comment by arunkumars08** [comment]
If that is the case, that's how it is currently happening. Maybe we can rethink on the delete icon to show a hide icon and clearly indicate it to the user via a tooltip.
